src/Todo/Model.hs

```diff
@@ -31,4 +49,32 @@ data Todo = Todo
     , todoSubtasks    :: Todos
     , todoDeadline    :: Maybe Deadline
     , todoDescription :: Maybe String
-    } deriving (Eq, Show)
+    } deriving (Eq, Show, Generic)
+
+instance Hashable Todo 
+
+defTodo :: Title -> Priority -> Tags -> Todo
+defTodo todoTitle todoPriority todoTags = Todo { todoState = TODO
+                                               , todoSubtasks = HSet.empty
+                                               , todoDeadline = Nothing
+                                               , todoDescription = Nothing
+                                               , .. }
+
+getByState :: State -> Todos -> Todos
+getByState state = HSet.filter $ (state ==) . todoState
+
+getByTag :: Tag -> Todos -> Todos
+getByTag tag = HSet.filter $ NESet.member tag . todoTags
+
+getTopPrio :: Int -> Todos -> Todos
+getTopPrio n = HSet.fromList . take n . sortBy (flip $ comparing todoPriority) . toList
+
+tagStatistics :: Todos -> HashMap Tag Int
+tagStatistics = count . concatMap (toList . todoTags)
+  where
+    count :: (Eq a, Hashable a) => [a] -> HashMap a Int
+    count [] = mempty
+    count (x:xs) = HMap.insert x (cnt + 1) mapp
+      where 
+        mapp = count xs
+        cnt = HMap.lookupDefault 0 x mapp
```

Comment:
Yeah, I can understand you here... Because of the fact that `map` is standard function name you need to use names like `mapp` or `mp`. I prefer `dict` as a shorten for _dictionary_ :slightly_smiling_face: